packages/seacas/libraries/nemesis/CMakeLists.txt

```diff
@@ -28,11 +28,14 @@ TRIBITS_ADD_LIBRARY(
   SOURCES ${SOURCES}
   )
 
+IF (${PACKAGE_NAME}_ENABLE_TESTS)
```

Comment:
I didn't think that this was needed if the `TRIBITS_ADD_EXECUTABLE_AND_TEST` function was used?

Reply:
Do you know what the command is to turn off nemesis tests from the cmake config? Something like `SEACASNemesis_ENABLE_TESTS=OFF`? I wasn't able to narrow it down earlier, so I added this. I can remove it for this PR, since these changes will remove the need for the if.

Reply:
> I didn't think that this was needed if the `TRIBITS_ADD_EXECUTABLE_AND_TEST` function was used?

As documented at the bottom of the section:

* https://tribitspub.github.io/TriBITS/users_guide/index.html#tribits-add-executable-and-test

> Finally, the tests are only added if tests are enabled for the package (i.e. [${PACKAGE_NAME}_ENABLE_TESTS](https://tribitspub.github.io/TriBITS/users_guide/index.html#package-name-enable-tests) = ON) and other criteria are met. But the test executable will always be added if this function is called, regardless of the value of `${PACKAGE_NAME}_ENABLE_TESTS`. To avoid adding the test (or example) executable when `${PACKAGE_NAME}_ENABLE_TESTS=OFF`, put this command in a subdir under test/ or example/ and that subdir with [tribits_add_test_directories()](https://tribitspub.github.io/TriBITS/users_guide/index.html#tribits-add-test-directories) or [tribits_add_example_directories()](https://tribitspub.github.io/TriBITS/users_guide/index.html#tribits-add-example-directories), respectively.

Reply:
OK, I misunderstood this then.  I thought I used to have these all inside if blocks so they were only built if tests were enabled and that the `TRIBITS_ADD_EXECUTABLE_AND_TEST` took care of that, so I removed the if blocks.  I will put those back so test executables are only built if tests are enabled...

Reply:
The origin of this problem is in Nalu-Wind. Our Cmake options are shown here:
https://github.com/sandialabs/spack-manager/blob/main/repos/exawind/packages/trilinos/package.py#L114
However, we still are forced to wrap the nemesis unit test in order to build, i.e.:
+IF (${PACKAGE_NAME}_ENABLE_TESTS)
 TRIBITS_ADD_EXECUTABLE_AND_TEST(
   ne_ctest_wrap
   NAME ne_ctest_wrap
   SOURCES ne_ctest_wrap.c
   NUM_MPI_PROCS 1
 )
+ENDIF()
Is there something wrong with the cmake options?

Reply:
> The origin of this problem is in Nalu-Wind. Our Cmake options are shown here: https://github.com/sandialabs/spack-manager/blob/main/repos/exawind/packages/trilinos/package.py#L114 However, we still are forced to wrap the nemesis unit test in order to build, i.e.: +IF (${PACKAGE_NAME}_ENABLE_TESTS) TRIBITS_ADD_EXECUTABLE_AND_TEST( ne_ctest_wrap NAME ne_ctest_wrap SOURCES ne_ctest_wrap.c NUM_MPI_PROCS 1 ) +ENDIF() Is there something wrong with our cmake options?

So is the problem that `ne_ctest_wrap` does not build (and perhaps adding the `LINKER_LANGUAGE` would fix it, or is there another reason that I am missing?

Reply:
The test fails to build because the linker is messed up with rdc. My question is whether we should expect
-DSEACASNemesis_ENABLE_TESTS=OFF
to work. If yes, then I think those changes I show above are necessary.

Reply:
NOTE: If only a small number of specific test executables don't build, you can add configure-time options to disable defining those executables by setting:

```
-D <exeTargetName>_EXE_DISABLE=ON
```

(see [here](https://tribits.org/doc/TribitsBuildReference.html#disabling-specific-test-executable-builds)).

But if the intent was to not build the test executables if `<Package>_ENABLE_TESTS=OFF`, then wrapping the calls in if() statements or (better) in a directory added using [`tribits_add_test_directories()`](https://tribitspub.github.io/TriBITS/users_guide/index.html#tribits-add-test-directories).

Reply:
That was the intent. 
https://github.com/trilinos/Trilinos/blob/master/packages/seacas/libraries/nemesis/CMakeLists.txt#L31
seems to wrap the test. 

However, it still gets compiled, despite having -DSEACASNemesis_ENABLE_TESTS=OFF, and then breaks because of the broken linker.